native/more/relationship-list.react.js

```diff
@@ -0,0 +1,100 @@
+// @flow
+
+import type { NavigationRoute } from '../navigation/route-names';
+import type { AppState } from '../redux/redux-setup';
+import type { MoreNavigationProp } from './more.react';
+
+import * as React from 'react';
+import { View, Text, FlatList } from 'react-native';
+
+import { connect } from 'lib/utils/redux-utils';
+import { type UserInfo } from 'lib/types/user-types';
+
+import { AddFriendsModalRouteName } from '../navigation/route-names';
+import { styleSelector } from '../themes/colors';
+
+import RelationshipListItem from './relationship-list-item.react';
+
+type Props = {|
+  navigation: MoreNavigationProp<>,
+  route: NavigationRoute<'FriendList' | 'BlockList'>,
+  // Redux state
+  styles: typeof styles,
+|};
+
+const DATA = [
+  { id: '1', username: 'John' },
+  { id: '2', username: 'Emma' },
+];
+
+class RelationshipList extends React.PureComponent<Props> {
+  render() {
+    return (
+      <View style={this.props.styles.container}>
+        <FlatList
+          contentContainerStyle={this.props.styles.contentContainer}
+          ListEmptyComponent={
```

Comment:
I thought we discussed the use of the inline definitions like this - they force the rerender of `FlatList` every time `RelationshipList` is rendered